### PR TITLE
Upgrade Datasette

### PR DESCRIPTION
**Pull request by earthboundkid**
Adds sorting by columns.

### CODE REVIEW

File: templates/database.html [hunk 1]
```diff
@@ -13,7 +13,7 @@
 <form class="sql" action="/{{ database }}-{{ database_hash }}" method="get">
     <h3>Custom SQL query</h3>
     <p>Note: Database uses <a href="https://www.sqlite.org/lang.html">SQLite</a> syntax.</p>
-    <p><textarea name="sql">select * from {{ tables[0].name|escape_table_name }}</textarea></p>
+    <p><textarea name="sql">select * from {{ tables[0].name|escape_sqlite }}</textarea></p>
```
Comment (earthboundkid) [inline templates/database.html:16]: `escape_table_name` was removed and replaced in the new Datasette.

File: templates/_rows_and_columns.html [hunk 2]
```diff
@@ -6,8 +6,17 @@
...
+                <th scope="col" class="{{ column.name }}">
+                    {% if not column.sortable %}
+                        {{ name }}
+                    {% else %}
+                        {% if column.name == sort %}
+                            <a href="{{ path_with_added_args(request, {'_sort_desc': column.name, '_sort': None, '_next': None}) }}" rel="nofollow">{{ name }}&nbsp;▼</a>
+                        {% else %}
+                            <a href="{{ path_with_added_args(request, {'_sort': column.name, '_sort_desc': None, '_next': None}) }}" rel="nofollow">{{ name }}{% if column.name == sort_desc %}&nbsp;▲{% endif %}</a>
+                        {% endif %}
+                    {% endif %}
```
Comment (earthboundkid) [inline templates/_rows_and_columns.html]: Most of this is copy-pasta from Datasette's new default template.

File: templates/500.html [hunk 3]
```diff
@@ -13,14 +13,16 @@
...
+        <p><textarea name="sql"></textarea></p>
 
-    <p><input type="submit" value="Run SQL"></p>
-</form>
+        <p><input type="submit" value="Run SQL"></p>
+    </form>
 
-{% include "_codemirror_foot.html" %}
+    {% include "_codemirror_foot.html" %}
+{% endif %}
```
Comment (earthboundkid) [inline templates/500.html:26]: Error handling was changed in the new Datasette.

**Comment by earthboundkid** [comment]
I found a bug with sorting NULL columns. It's reported here: https://github.com/simonw/datasette/issues/216

We should probably wait until that's worked out before upgrading.

**Comment by simonw** [comment]
I just shipped Datasette 0.19 with a fix for https://github.com/simonw/datasette/issues/216

**Comment by earthboundkid** [comment]
Thanks, Simon!

**Pull request merged**